client/controller/index.web.js

```diff
@@ -39,9 +40,16 @@ export const ProviderWrappedLayout = ( {
 } ) => {
 	const state = store.getState();
 	const userLoggedIn = isUserLoggedIn( state );
+	const [ , experimentAssignment ] = useExperiment(
+		'calypso_sidebar_upsell_dedicated_upgrade_flow_202301'
+	);
 
 	const layout = userLoggedIn ? (
-		<Layout primary={ primary } secondary={ secondary } />
+		<Layout
+			primary={ primary }
+			secondary={ secondary }
+			experimentAssignment={ experimentAssignment }
```

Comment:
Is `experimentAssignment` really necessary to provide this so early? Can we inject it where it's actually needed?

Reply:
Good point @tyxla, [we fixed this behavior here](https://github.com/Automattic/wp-calypso/pull/72932/commits/8467502c2827e3d4a308f65763af209fe915a728)